sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -102,6 +102,12 @@ case class DropDatabaseCommand(
   extends RunnableCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    if (cascade) {
+      catalog.listTables(databaseName).foreach { t =>
+        catalog.refreshTable(t)
+      }
+    }
     sparkSession.sessionState.catalog.dropDatabase(databaseName, ifExists, cascade)
```

Comment:
On this call we can get an error. For example, its not allowed to drop the default database. In that case, even though we are not going to go ahead with the drop database action, we will end up refreshing all the tables inside it ? Is that expected ?